wsgi.py

```diff
@@ -1,55 +1,73 @@
 import os
 import logging
+import io
+import tarfile
+import tempfile
+import json
+import requests
 
 from flask import Flask, jsonify, request
 from flask.logging import default_handler
 
-import s3
-import producer
-
 application = Flask(__name__)  # noqa
 
 # Set up logging
 ROOT_LOGGER = logging.getLogger()
 ROOT_LOGGER.setLevel(application.logger.level)
 ROOT_LOGGER.addHandler(default_handler)
 
-# Kafka message bus
-SERVER = os.environ.get('KAFKA_SERVER')
-TOPIC = os.environ.get('KAFKA_TOPIC')
-
-# S3 credentials
-AWS_KEY = os.environ.get('AWS_ACCESS_KEY_ID')
-AWS_SECRET = os.environ.get('AWS_SECRET_ACCESS_KEY')
-AWS_BUCKET = os.environ.get('AWS_S3_BUCKET_NAME')
+# Upload Service
+UPLOAD_SERVICE_ENDPOINT = os.environ.get('UPLOAD_SERVICE_ENDPOINT')
 
 
 @application.route("/", methods=['POST'])
 def wake_up():
     """Endpoint for upload and publish requests."""
     input_data = request.get_json(force=True)
     data_id = input_data['id']
-    ai_service_id = input_data.get('ai_service', 'generic_ai')
     raw_data = input_data['data']
 
-    s3_destination = f'{AWS_BUCKET}/{data_id}/{ai_service_id}'
-
-    application.logger.info(
-        'Saving data to location:  s3://%s', s3_destination
-    )
-
-    filesystem = s3.connect(AWS_KEY, AWS_SECRET)
-    s3.save_data(filesystem, s3_destination, raw_data)
-
-    message = {
-        'message': f'AI-Ops pipeline successfull for {data_id}',
-        'url': s3_destination
+    try:
+        # create a tar.gz file using a temp file
+        with tempfile.NamedTemporaryFile(dir='/tmp', delete=False) as tmp_file:
+            temp_file_name = tmp_file.name
+            with tarfile.open(temp_file_name, "w:gz") as tar:
+                data = io.BytesIO()
+                data_len = data.write(json.dumps(raw_data).encode())
+                info = tar.tarinfo()
+                info.name = 'aiservice_id_' + data_id
+                info.size = data_len
+
+                # add the tar.gz file to tar and close it
+                data.seek(0)
+                tar.addfile(info, data)
+                tar.close()
+    except IOError as e:
+        return jsonify(
```

Comment:
This return statement would respond to the client with `HTTP 200 SUCCESS` status. Returning a JSON stating that something bad happened is not enough. See: http://flask.pocoo.org/docs/1.0/quickstart/#about-responses

```suggestion
        return jsonify(...), 500
```